backend/app/api/endpoints/adapter/attachments.py

```
@@ -65,23 +65,31 @@ async def upload_attachment(
+        CHUNK_SIZE = 1024 * 1024  # 1MB chunks
+        chunks = []
+        file_size = 0
+        while chunk := await file.read(CHUNK_SIZE):
+            file_size += len(chunk)
+            # Check size during streaming to fail fast
+            if not DocumentParser.validate_file_size(file_size):
+                max_size_mb = DocumentParser.get_max_file_size() / (1024 * 1024)
+                raise HTTPException(
+                    status_code=400,
+                    detail=f"File size exceeds maximum limit ({max_size_mb} MB)",
+                )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)